pinot-common/src/main/java/org/apache/pinot/common/utils/TarCompressionUtils.java

```diff
@@ -66,106 +67,122 @@ public class TarGzCompressionUtils {
    */
 
 
-  private TarGzCompressionUtils() {
+  private TarCompressionUtils() {
   }
 
   public static final String TAR_GZ_FILE_EXTENSION = ".tar.gz";
+  public static final String TAR_LZ4_FILE_EXTENSION = ".tar.lz4";
+  public static final String TAR_ZST_FILE_EXTENSION = ".tar.zst";
+  public static final Map<String, String> COMPRESSOR_NAME_BY_FILE_EXTENSIONS = Map.of(
+      TAR_GZ_FILE_EXTENSION, CompressorStreamFactory.GZIP,
+      TAR_LZ4_FILE_EXTENSION, CompressorStreamFactory.LZ4_FRAMED,
+      TAR_ZST_FILE_EXTENSION, CompressorStreamFactory.ZSTANDARD
+  );
+  private static final CompressorStreamFactory COMPRESSOR_STREAM_FACTORY = new CompressorStreamFactory();
```

Comment:
There is already a singleton defined in `CompressorStreamFactory`

Reply:
Switch to using singleleton